R/modspec.R

```diff
@@ -195,24 +195,48 @@ check_spec_contents <- function(x, crump = TRUE, warn = TRUE, ...) {
   return(invisible(NULL))
 }
 
-audit_spec <- function(x, spec, warn = TRUE) {
-  cmt <- names(init(x))
-  if(!has_name("ODE", spec) | !warn | length(cmt) ==0) {
+grepl_dxdt_ode <- function(spec, cmt) {
+  dx <- paste0("dxdt_", cmt)
+  z <- vapply(dx, FUN.VALUE = TRUE, function(dxi) {
+    any(grepl(dxi, spec[["ODE"]], fixed = TRUE))
```

Comment:
Are partial matches here a concern?  For example

```diff
diff --git a/inst/maintenance/unit/test-audit.R b/inst/maintenance/unit/test-audit.R
index 482f0ec1..1897b667 100644
--- a/inst/maintenance/unit/test-audit.R
+++ b/inst/maintenance/unit/test-audit.R
@@ -38,4 +38,10 @@ test_that("test audit ode block dadt", {
   code6 <- "$CMT @number 3"
   expect_silent(mod <- mcode("audit6", code6, compile = FALSE))
   expect_is(mod, "mrgmod")
+
+  code7 <- "$CMT A \n $ODE \n dxdt_A1  = 0;"
+  expect_warning(
+    mcode("audit7", code7, compile = FALSE, quiet = FALSE), 
+    regexp = "--| missing: dxdt_A"
+  )
 })
```

Perhaps something like this to make it more specific:

```diff
diff --git a/R/modspec.R b/R/modspec.R
index c0be7196..12b1a0c4 100644
--- a/R/modspec.R
+++ b/R/modspec.R
@@ -198,7 +198,8 @@ check_spec_contents <- function(x, crump = TRUE, warn = TRUE, ...) {
 grepl_dxdt_ode <- function(spec, cmt) {
   dx <- paste0("dxdt_", cmt)
   z <- vapply(dx, FUN.VALUE = TRUE, function(dxi) {
-    any(grepl(dxi, spec[["ODE"]], fixed = TRUE))
+    pat <- paste0(" *\\Q", dxi, "\\E", " *=")
+    any(grepl(pat, spec[["ODE"]]))
   })  
   names(z) <- dx
   z
```

Reply:
Good idea; thanks. Tests added for this as well.

Reply:
I never came across `\Q ... \E` before; pretty cool; will definitely use this.